zerver/lib/actions.py

```diff
@@ -1753,6 +1755,9 @@ def do_remove_reaction(user_profile: UserProfile, message: Message,
                                        message=message,
                                        emoji_code=emoji_code,
                                        reaction_type=reaction_type).get()
+    if len(Reaction.objects.filter(message=message)) <= 1:
+        message.has_reaction = False
+        message.save(update_fields=["has_reaction"])
```

Comment:
Let's do this one line down and check `not Reaction.objects.filter(message=message).exists()`